app/policies/rating_policy.rb

```diff
@@ -0,0 +1,9 @@
+class RatingPolicy < ApplicationPolicy
+  def create?
+    user.present? && Rating.find_by(article: record.article, user: record.user).blank?
```

Comment:
should we use `before_action` hook in controllers to avoid checking user existence?
ref: https://github.com/heartcombo/devise#controller-filters-and-helpers
```ruby
# in controller
before_action :authenticate_user!
```

Reply:
should we use constrains on DB and model level instead of using `Rating.find_by(article: record.article, user: record.user).blank?` contruction? 
I mean we should add SQL constraint to checking the uniqueness by `[ratings.article_id, ratings.user_id], also add validation to model